doc/scenario.md

```diff
@@ -0,0 +1,317 @@
+# Role Scenario
+
+The role uses a scenario of steps to determine what to do.
+
+Using a scenario of steps, you can:
+- run the specified steps in any order, but API must be followed (with `cartridge_scenario`);
+- replace the steps of the role with your own or add new steps
+  (with `cartridge_custom_steps_dir` or `cartridge_custom_steps`).
+
+## Steps
+
+In `cartridge_scenario` you should specify names of steps. The default scenario includes the following steps:
+- [deliver_package](#deliver_package)
+- [update_package](#update_package)
+- [update_instance](#update_instance)
+- [configure_instance](#configure_instance)
+- [restart_instance](#restart_instance)
+- [wait_instance_started](#wait_instance_started)
+- [connect_to_membership](#connect_to_membership)
+- [edit_topology](#edit_topology)
+- [cleanup_expelled](#cleanup_expelled)
+- [configure_auth](#configure_auth)
+- [configure_app_config](#configure_app_config)
+- [bootstrap_vshard](#bootstrap_vshard)
+- [configure_failover](#configure_failover)
+
+In addition to the steps in the default scenario, there are additional steps:
+- [set_control_instance](#set_control_instance)
+
+To replace the steps of the role with your own or add new steps, you should use `cartridge_custom_steps_dir`
+or `cartridge_custom_steps` options.
+
+## Example
+
+For example, you can run `edit_topology` step without `connect_to_membership` step (for existing cluster):
+```yaml
+cartridge_scenario:
+  - edit_topology
+  - cleanup_expelled
+```
+
+Also, you can specify control instance in config for `edit_topology` step:
+```yaml
+control_instance:
+  name: core_1
+  console_sock: '/var/run/tarantool/core_1.control'
+```
+
+If you want to replace `deliver_package` step with a custom one, you should set a path to a directory
```

Comment:
### Replace copying package to remote host 

By default, package specified by `cartridge_package_path` is simple copied to remote host, it's happened on `deliver_package` step.
But sometimes we need to download or copy package from some custom place. Let's to it!

Generally, we just need to override default `deliver_package` step.
Let's take a look on it's API. It requires two variables - `cartridge_package_path` that we simple specify in vars, and `single_instances_for_each_machine` that allows us to run this task once per each machine. As a result of this module we should set `delivered_package_path` variable (a path of package on remote machine).

Now, choose a directory where our custom steps are placed, for example `custom_steps`.

Then create `custom_steps/deliver_package.yml` and describe a way to deliver package:

```yaml
---

- when:
    - cartridge_package_path is not none
    - inventory_hostname in single_instances_for_each_machine
  tags: cartridge-instances
  block:
      get_url:
        url: 'https://some-server/myapp-1.0.0-0.rpm'
        dest: '/tmp'
     register: downloaded_package

    - name: 'Set delivered package path'
      set_fact:
        delivered_package_path: '{{ downloaded_package.dest }}'
```

Import role and say where to find our custom steps:

```yaml
# deploy_application.yml

- name: My playbook
  hosts: all
  tasks:
    - name: Deploy application
      import_role:
        name: tarantool.cartridge
      vars:
        cartridge_custom_steps_dir: "./custom_steps"
```